wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -24,6 +25,41 @@ import scala.util.{Failure, Success, Try}
   */
 private[wallet] trait TransactionProcessing extends WalletLogger {
   self: Wallet =>
+
+  private case object RebroadcastTransactionsRunnable extends Runnable {
+
+    override def run(): Unit = {
+      while (!rebroadcastTxsThread.isInterrupted) {
+        val f = for {
+          txs <- getTransactionsToBroadcast
+          _ <- nodeApi.broadcastTransactions(txs)
+        } yield ()
+
+        // Make sure broadcasting completes
+        // Wrap in try in case of spurious failure
+        Try(Await.result(f, 60.seconds))
+
+        // wait 30 minutes
+        Thread.sleep(30.minutes.toSeconds)
+      }
+    }
+  }
+
+  private lazy val rebroadcastTxsThread = new Thread(
```

Comment:
So instead of allocating a new thread, i think we should just create a wallet scheduler just like we have with hikari

https://github.com/bitcoin-s/bitcoin-s/blob/5b762fb2e13742f057162e14a035b67730053ca0/db-commons/src/main/scala/org/bitcoins/db/HikariLogging.scala#L181

We may even be able to re-use this scheduler for the address queue stuff as well? 

I guess I don't want to keep allocating single threads for various functions in the wallet when realistically all these things could share the same background thread. Thoughts?